controllers/reconcile_persistence.go

```diff
@@ -0,0 +1,177 @@
+package controllers
+
+import (
+	"errors"
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	rabbitmqv1beta1 "github.com/rabbitmq/cluster-operator/api/v1beta1"
+	"github.com/rabbitmq/cluster-operator/internal/resource"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	k8sresource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+
+func (r *RabbitmqClusterReconciler) reconcilePVC(ctx context.Context, builder resource.ResourceBuilder, cluster *rabbitmqv1beta1.RabbitmqCluster, resource client.Object) error {
+	logger := ctrl.LoggerFrom(ctx)
+
+	switch sts := resource.(type) {
+	case *appsv1.StatefulSet:
+		current, err := r.statefulSet(ctx, cluster)
+		if client.IgnoreNotFound(err) != nil {
+			return err
+		} else if k8serrors.IsNotFound(err) {
```

Comment:
I was thinking about that, but personally I found listed the error NotFound as a if condition to be more specific. Without it listed here, I might need to stop and think about what's going to happen to this case. I would prefer to leave it as it is, but not a strong opinion ;) How strongly do you feel about this?

Reply:
Giving it a second thought, you do need both `if` statements because the error could be `nil` because there was no error at all. A simple `else` without further check won't evaluate this case correctly.